packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -3563,6 +3564,83 @@ export default abstract class SqlIntegration
 `;
     return format(sqlQuery, this.getFormatDialect());
   }
+
+  factMetricAlreadyExists(
+    existingFactMetrics: FactMetricInterface[],
+    autoFactMetric: AutoFactMetricToCreate
+  ): boolean {
+    return existingFactMetrics.some((existing) => {
+      //TODO: Improve this logic
+      return (
+        existing.metricType === autoFactMetric.metricType &&
+        existing.numerator.column === autoFactMetric.numerator.column
+      );
+    });
+  }
+
+  getAutoFactMetricsToCreate(
+    existingFactMetrics: FactMetricInterface[],
+    factTable: FactTableInterface
+  ): AutoFactMetricToCreate[] {
+    const event = factTable.eventName;
+    // based on these docs - https://support.google.com/analytics/answer/9267735?hl=en&ref_topic=13367566&sjid=16477269010846037969-NA
+    const ga4InvertedEvents: string[] = [
```

Comment:
TODO: Make this more generic and include recommended events from Rudderstack/Segment/Amplitude